File: Level1_K번째수.py
def solution(array, commands):
    answer = []
    stack = []

    for i in commands:
        # commands 내의 1번째 list 값에서 [2, 5]를 가져와서 인덱스로 사용
        # 즉, array[2 : 5]가 되는데 array의 2번째부터 5번째를 자르면
        # [5, 2, 6, 3]이다. 하지만 배열은 인덱스가 0부터 시작하므로,
        # 실제 2번째 값은 array[1]이므로, -1을 해줘야 한다.
        stack = array[i[0] - 1: i[1]]
        stack.sort()

        # 마찬가지 이유로 -1을 했다.
        answer.append(stack[i[2] - 1])

    return answer

File: C++/Level1_K번째 수.cpp
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> array, vector<vector<int>> commands) {
	vector<int> answer;

	// 자르기
	for (int i = 0; i < commands.size(); i++)
	{
		// 임시 벡터를 생성 
		vector<int> temp;

		// 생성한 임시 벡터에 array[i] 부터 array[j] 까지 원소를 저장.
		// -1을 한 이유는 문제에서 원소 인덱스보다 1이 더 크므로 -1을 해준다.
		// -1을 안하게 되면 2, 6, 3만 저장될 것임.
		for (int j = commands[i][0] - 1; j < commands[i][1]; j++)
			temp.push_back(array[j]);

		// 임시 벡터 정렬
		sort(temp.begin(), temp.end());

		// k에 해당하는 인덱스 원소 저장
		// -1을 한 이유는 위의 for문에서 한 것과 동일하다. 
		// -1을 안하면 목표한 인덱스의 다음 인덱스를 참조하여 answer에 저장하게 되어버린다.
		answer.push_back(temp[commands[i][2] - 1]);
	}
	return answer;
}